libs/hwes/aggregate.go

```diff
@@ -0,0 +1,196 @@
+package hwes
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+)
+
+type eventHandler func(evt Event) error
+
+type AggregateType string
+
+// Aggregate reflects a concept of Domain-Driven-Design.
+// This interface describes the contracts that are used
+// by an AggregateStore to persists this aggregate in an
+// event-sourcing manner
+type Aggregate interface {
+	GetID() uuid.UUID
+	GetStreamID() string // TODO: Make private https://github.com/helpwave/services/pull/587#discussion_r1451788115
```

Comment:
this TODO is still open

Reply:
Done. I reframed the function to respect separation of concerns.

Reply:
Maybe GitHub shows me an old version (can't really check rn, am on mobile), but this comment still exists?